src/ralph/backends/data/clickhouse.py

```diff
@@ -0,0 +1,524 @@
+            # succeeded, we assume 0 here.
+            return 0
+        inserted_count = len(batch)
+        logger.debug("Inserted %d documents chunk with success", inserted_count)
+
+        return inserted_count
+
+    def write(  # pylint: disable=too-many-arguments disable=too-many-branches
+        self,
+        data: Union[IOBase, Iterable[Union[bytes, dict]]],
+        target: Union[None, str] = None,
+        chunk_size: Union[None, int] = None,
+        ignore_errors: bool = False,
+        operation_type: Union[None, BaseOperationType] = None,
+    ) -> int:
+        """Writes documents from the `stream` to the instance table.
+
+        Args:
+            data: The data to write to the database.
+            target: The target table to write to.
+            chunk_size: The number of documents to write at once.
+            ignore_errors: Whether to ignore errors or not.
+            operation_type: The operation type to use for the write operation.
+        """
+        target = target if target else self.event_table_name
+        if not operation_type:
+            operation_type = self.default_operation_type
+        if not chunk_size:
+            chunk_size = self.default_chunk_size
+        logger.debug(
+            "Start writing to the %s table of the %s database (chunk size: %d)",
+            target,
+            self.database,
+            chunk_size,
+        )
+
+        data = iter(data)
+        try:
+            first_record = next(data)
+            data = chain([first_record], data)
+            if isinstance(first_record, bytes):
+                data = self._parse_bytes_to_dict(data, ignore_errors)
+        except StopIteration:
+            logger.info("Data Iterator is empty; skipping write to target.")
+            return 0
+
+        success = 0
+        batch = []
+        if operation_type in [BaseOperationType.CREATE]:
+            for document in self.to_documents(
+                data, ignore_errors=ignore_errors, operation_type=operation_type
+            ):
+                batch.append(document)
+                if len(batch) < chunk_size:
+                    continue
+
+                success += self.bulk_import(
+                    batch,
+                    ignore_errors=ignore_errors,
+                    event_table_name=target,
+                )
+                batch = []
+
+            # Edge case: if the total number of documents is lower than the chunk size
+            if len(batch) > 0:
+                success += self.bulk_import(
+                    batch,
+                    ignore_errors=ignore_errors,
+                    event_table_name=target,
+                )
+
+            logger.debug("Inserted a total of %d documents with success", success)
+        else:
+            msg = "%s operation_type is not allowed."
+            logger.error(msg, operation_type.name)
+            raise BackendParameterException(msg % operation_type.name)
```

Comment:
`UPDATE` operation type has been postponed or will never be supported? Also, what about `DELETE`?